diplom.py

```diff
@@ -52,33 +61,31 @@ def mean_average_precision(predicted_probability_distribution,
 # print train_X[0], train_y[0]
 # print test_X[0], test_y[0]
 
-from sklearn import linear_model
-logistic = linear_model.LogisticRegression(C = 1, solver = 'lbfgs', 
-    multi_class = 'multinomial', max_iter = 100)
-print('LogisticRegression score: %f'
-      % logistic.fit(train_X, train_y).score(test_X, test_y))
+
+if args.model == "logistic_regression":
+    logistic = linear_model.LogisticRegression(C = 1, solver = 'lbfgs', 
+                                               multi_class = 'multinomial', max_iter = 100)
+else:
+    logistic = RandomForestClassifier(n_estimators=100)
+
+logistic.fit(train_X, train_y)
+
 
 predicted_probability_distribution_test = logistic.predict_proba(test_X)
 predicted_probability_distribution_train = logistic.predict_proba(train_X)
 
-
 MAP_test = mean_average_precision(predicted_probability_distribution_test, 
                                   logistic.classes_, 
                                   test_y)
-print MAP_test
+print "MAP on test data: %s" % MAP_test
```

Comment:
Although, this will work because of implicit type conversion of float to string, I would suggest you to format that float by hand and limit precision by changing "%s" to "%.3f".